File: Atcoder_#295/javascript/c_socks.js
function main(input) {
    const lines = input.trim().split('\n');  // 入力を行ごとに分割
    const n = parseInt(lines[0], 10);  // Nを取得
    const socks = lines[1].split(' ').map(Number);  // 靴下の色の配列を取得

    const sockCount = {};  // 色ごとの靴下の数を数えるオブジェクト

    // 靴下の色ごとにカウント
    for (const sock of socks) {
        sockCount[sock] = (sockCount[sock] || 0) + 1;
    }

    let pairs = 0;

    // 各色ごとの靴下の数を2で割ってペアを計算
    for (const count of Object.values(sockCount)) {
        pairs += Math.floor(count / 2);
    }

    console.log(pairs);  // 結果を出力
}

// 標準入力を読み込み、main関数を呼び出す処理
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8');
main(input);

File: Atcoder_#295/ruby/c_socks.rb
def main
  # N を取得
  n = gets.to_i
  # 靴下の色を表す配列 A を取得
  a = gets.split.map(&:to_i)

  # 色ごとの靴下の数をカウントするためのハッシュ
  sock_count = Hash.new(0)

  # 靴下の色ごとにカウント
  a.each do |color|
    sock_count[color] += 1
  end

  # ペアの数を数える
  pairs = 0
  sock_count.each_value do |count|
    pairs += count / 2  # 2枚で1ペア
  end

  # 結果を出力
  puts pairs
end

main
